### ISSUE TITLE
Mint support

### ISSUE THREAD

**Issue by LPGhatguy** [rustgd/glsl-layout#7]
In order to support all the math libraries, it would be useful if glsl-layout would support conversions from any type that can be converted into the corresponding mint type.

I put together something similar for the (flawed) [shader-types crate](https://github.com/BVE-Reborn/shader-types), but with a derive macro I think the ergonomics can be even better.

**Comment by zakarumych** [comment]
Yes, this sounds useful. I would be glad to merge a PR with this :)
